gdal/alg/gdal_rpc.cpp

```diff
@@ -1909,7 +1924,16 @@ static bool GDALRPCOpenDEM( GDALRPCTransformInfo* psTransform )
         psTransform->nBufferHeight = -1;
         psTransform->nLastQueriedX = -1;
         psTransform->nLastQueriedY = -1;
-        auto poDSSpaRefSrc = psTransform->poDS->GetSpatialRef();
+
+        OGRSpatialReference* poDEMSRS = nullptr;
```

Comment:
this will leak memory. Just declare `OGRSpatialReference oDEMSRS;`

Reply:
When I make this change I get a complaint at gdal_rpc:1936
```
gdal_rpc.cpp: In function ‘bool GDALRPCOpenDEM(GDALRPCTransformInfo*)’:
gdal_rpc.cpp:1936:64: error: operands to ?: have different types ‘OGRSpatialReference’ and ‘const OGRSpatialReference*’
 1936 |         auto poDSSpaRefSrc = psTransform->pszDEMSRS != nullptr ? poDEMSRS : psTransform->poDS->GetSpatialRef();
      |                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gdal_rpc.cpp:2026:20: error: type ‘<type error>’ argument given to ‘delete’, expected pointer
 2026 |             delete poDSSpaRef;
```

and my intellisense says that there is assignment between incompatible types `OGRSpatialReference` and `OGRSpatialReference *`. I can see the issue but I'm not sure how to resolve it?